### ISSUE TITLE
Encoding strings doesn't work

### ISSUE THREAD

**Issue by alfador1** [linkchecker/linkchecker#533]
## Summary
Text not encoded correctly.
I have set the locales correctly:
/opt/linkchecker # locale -a
C
C.UTF-8
nb_NO.UTF-8
en_GB.UTF-8
en_US.UTF-8
es_ES.UTF-8
it_IT.UTF-8
ru_RU.UTF-8
fr_FR.UTF-8
ch_DE.UTF-8
pt_BR.UTF-8
sv_SE.UTF-8
de_DE.UTF-8
nl_NL.UTF-8
de_CH.UTF-8

I keep receiving malformed text like:

[Bes√∂k oss p√•
 
Beg√§r   en offert     f√∂r DHL Express
Klicka   h√§r f√∂r att g√• till Freight     DHL Active Tracing

## Steps to reproduce

 1.  linkchecker --no-robots --output=csv/utf_8 --check-extern https://www.dhl.com/se-sv/home.html | tee Test.csv
 2. Check the CSV and see that the name column contains malformed names.

## Expected result
Properly encoded text
## Environment

 * Operating system: 
 * Linkchecker version: 
 * Python version: 3.9.1
 * Install method: from source

**Comment by cjmayo** [comment]
I've identified a LinkChecker problem and a solution in #572 for http(s) links. Testing welcome. I think that is all LinkChecker can do - it may not be sufficient.

Even without that commit, checking the page as a local file from linkchecker (i.e. not via http) can help debug your system. If it still doesn't get the encoding right, do you have chardet installed? Which version?

Then one further option:
- install [cchardet](https://pypi.org/project/cchardet/) #571

**Comment by lpirl** [comment]
Using LinkChecker 10.0.1 on Python 3.9 here, no matter if `cchardet` installed or not, no matter if directory or via http ([e.g., hosted here](https://static.lukas-pirl.de/linkchecker-utf8/)), I cannot get rid of false positives checking the following pages:

`index.html`:
``` html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8" />
  <title>linkchecker utf-8 test, index page</title>
</head>
<body>
  index page,
  <a href="ö.html">link to ö page</a>,
  <a href="🍞.html">link to 🍞 page</a>
</body>
</html>
```

`ö.html`:
``` html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8" />
  <title>linkchecker utf-8 test, ö page</title>
</head>
<body>
  ö page,
  <a href="index.html">link to index page</a>,
  <a href="🍞.html">link to 🍞 page</a>
</body>
</html>
```

`🍞.html`:
``` html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8" />
  <title>linkchecker utf-8 test, 🍞 page</title>
</head>
<body>
  🍞 page,
  <a href="index.html">link to index page</a>,
  <a href="ö.html">link to ö page</a>,
</body>
</html>
```

**Comment by cjmayo** [comment]
@lpirl are we saying broken on 10.0.1 365852a4428 fixed with #572? Works for me with http and local files.

**Comment by lpirl** [comment]
Correct, broken for me with 10.0.1, fixed for me with #572 – whatever the difference to your tests might be, hm.

**Comment by cjmayo** [comment]
572 merged.

@alfador1 do see #554. Thanks for your report - you were first and it did help prompt the investigation. We just seem to have written more on the other ticket.
